### PR TITLE
fix(manager, slice-machine-ui): compound M1 review

### PR DESCRIPTION
**Pull request by lihbr**
## Context

SMX-82, SMX-97, SMX-105

## Checklist before requesting a review
- [ ] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [ ] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: packages/manager/src/managers/SliceMachineManager.ts [hunk 1]
```diff
@@ -173,42 +174,60 @@ export class SliceMachineManager {
...
 			this.project.detectPackageManager(),
 		]);
 
-		// TODO: SM UI detects if a user is logged out by looking at
+		// SM UI detects if a user is logged out by looking at
 		// `clientError`. Here, we simulate what the old core does by
-		// returning an `ErrorWithStatus`-like object if the user is
-		// not logged in.
+		// returning an `ErrorWithStatus`-like object if the user does
+		// not have acces to the repository or is not logged in.
```
Comment (bapmrl) [inline packages/manager/src/managers/SliceMachineManager.ts]: ⚠️ Small typo.

File: packages/manager/src/managers/SliceMachineManager.ts [hunk 2]
```diff
@@ -173,42 +174,60 @@ export class SliceMachineManager {
...
+							remoteSlices,
+							unauthorizedError,
+						};
+					} catch (error) {
+						// Non-Prismic error
+						if (!(error instanceof PrismicError)) {
+							throw error;
+						}
+
+						unauthorizedError = {
```
Comment (bapmrl) [inline packages/manager/src/managers/SliceMachineManager.ts]: ❓ Does it really always correspond to an unauthorized error? What if it's an internal server error for instance?

File: packages/manager/src/managers/SliceMachineManager.ts (hunk 1 above)
Comment (lihbr) [reply packages/manager/src/managers/SliceMachineManager.ts]: Nice catch!

File: packages/manager/src/managers/SliceMachineManager.ts (hunk 2 above)
Comment (lihbr) [reply packages/manager/src/managers/SliceMachineManager.ts]: Basically, the API doesn't return a 401 upon unauthorized calls, it returns a 500, which makes triaging errors quite tricky :c

Here are the case we handle: https://github.com/prismicio/prismic-custom-types-client/blob/main/src/client.ts#L430-L497

Let me know if you have an idea on how to improve that 🤔

Comment (lihbr) [reply packages/manager/src/managers/SliceMachineManager.ts]: OK, now it actually returns 401, I'll check to update that on the remote client 👌

Comment (lihbr) [reply packages/manager/src/managers/SliceMachineManager.ts]: Done in [d252b4e](https://github.com/prismicio/slice-machine/pull/887/commits/d252b4e1778abb413fa5b4248744cdaf384e5725), let me know!

**Review by bapmrl — APPROVED** [review]

**Pull request merged**